src/game.cpp

```diff
@@ -1701,6 +1701,84 @@ ReturnValue Game::internalTeleport(Thing* thing, const Position& newPos, bool pu
 }
 
 //Implementation of player invoked events
+void Game::playerEquipItem(uint32_t playerId, uint16_t spriteId)
```

Comment:
As mentioned in a comment, I'd rather see a simpler implementation for this, like in https://gist.github.com/ninjalulz/ec6a8ef25072482ea688. Just find an item with the client id and move it to the appropriate slot using internalMoveItem.